test/integ/test_easy_logging.py

```diff
@@ -0,0 +1,71 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+
+from __future__ import annotations
+
+import os
+from logging import getLogger
+from pathlib import Path
+from test.integ.conftest import create_connection
+
+import pytest
+
+pytestmark = pytest.mark.skipolddriver
+try:
+    import tomlkit
+
+    from snowflake.connector.config_manager import CONFIG_MANAGER
+    from snowflake.connector.constants import CONFIG_FILE
+except ModuleNotFoundError:
+    pass
+
+
+@pytest.fixture(scope="function")
+def log_directory(tmp_path_factory):
+    return tmp_path_factory.mktemp("log")
+
+
+@pytest.fixture(scope="function")
+def temp_config_file(tmp_path_factory):
+    return tmp_path_factory.mktemp("config_file_path") / "config.toml"
+
+
+@pytest.fixture(scope="function")
+def config_file_setup(request, temp_config_file, log_directory):
+    param = request.param
+    CONFIG_MANAGER.file_path = Path(temp_config_file)
+    configs = {
+        "save_logs": {"log": {"save_logs": True, "path": str(log_directory)}},
+        "no_save_logs": {"log": {"save_logs": False, "path": str(log_directory)}},
+    }
+    try:
+        # create temp config file
+        with open(temp_config_file, "w") as f:
+            f.write(tomlkit.dumps(configs[param]))
+        yield
+    finally:
+        # remove created dir and file, including log paths and config file paths
+        CONFIG_MANAGER.file_path = CONFIG_FILE
+
+
+@pytest.mark.parametrize("config_file_setup", ["save_logs"], indirect=True)
+def test_save_logs(db_parameters, config_file_setup, log_directory):
+    create_connection("default")
+
+    assert os.path.exists(os.path.join(log_directory, "python-connector.log"))
+    with open(os.path.join(log_directory, "python-connector.log")) as f:
+        data = f.read()
+        assert "Snowflake Connector for Python" in data
```

Comment:
Probably better to move the `assert` outside of the `with` statement.

Reply:
will do